### PR TITLE
Added Oculus Manifest generation tool for Quest

### PR DESCRIPTION
**Pull request by SimonDarksideJ**
## Overview

Added Oculus Manifest generation tool for Quest

For the Quest to detect the controllers properly, a custom AndroidManifest.xml file is required.  Ported the Oculus tools to enable this for XRTK projects.

In the future, this will be handled by the Build Window

## Changes:

- Fixes: #13 

## Notes

A manual task, users will need to use the "Mixed Reality Toolkit -> Tools" options to generate the manifest. Has safety features to check if there is an existing file and won't overwrite until it's removed.

### CODE REVIEW

File: XRTK.Oculus/Packages/com.xrtk.oculus/Editor/OculusManifestPreprocessor.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+using UnityEngine;
+using UnityEditor;
+using System.IO;
+
+namespace XRTK.Oculus
+{
+	public class OculusManifestPreprocessor
+	{
+		[MenuItem("Mixed Reality Toolkit/Tools/Create store-compatible AndroidManifest.xml", false, 100000)]
```
Comment (StephenHodgson) [inline XRTK.Oculus/Packages/com.xrtk.oculus/Editor/OculusManifestPreprocessor.cs]: Not sure I want to add this to the menu or actually put it in the Build Window itself

Comment (StephenHodgson) [reply XRTK.Oculus/Packages/com.xrtk.oculus/Editor/OculusManifestPreprocessor.cs]: Also this doesn't specify that it's an Oculus specific method in the name of the menu item.

File: XRTK.Oculus/Packages/com.xrtk.oculus/Editor/OculusUpdaterStub.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+limitations under the License.
+
+************************************************************************************/
+
+using UnityEngine;
+using System.Collections;
+
+namespace XRTK.Oculus
+{
+	public class OculusUpdaterStub : ScriptableObject
```
Comment (StephenHodgson) [inline XRTK.Oculus/Packages/com.xrtk.oculus/Editor/OculusUpdaterStub.cs]: We can safely make this internal without breaking the `FromScriptableObject` lookup

File: XRTK.Oculus/Packages/com.xrtk.oculus/Editor/OculusManifestPreprocessor.cs [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+		public static void GenerateManifestForSubmission()
+		{
+			var so = ScriptableObject.CreateInstance(typeof(OculusUpdaterStub));
+			var script = MonoScript.FromScriptableObject(so);
+			string assetPath = AssetDatabase.GetAssetPath(script);
+			string editorDir = Directory.GetParent(assetPath).FullName;
+			string srcFile = editorDir + "/AndroidManifest.OVRSubmission.xml";
```
Comment (StephenHodgson) [inline XRTK.Oculus/Packages/com.xrtk.oculus/Editor/OculusManifestPreprocessor.cs]: let's use string interpolation where possible.

File: XRTK.Oculus/Packages/com.xrtk.oculus/Editor/OculusManifestPreprocessor.cs (hunk 1 above)
Comment (SimonDarksideJ) [reply XRTK.Oculus/Packages/com.xrtk.oculus/Editor/OculusManifestPreprocessor.cs]: Updated

File: XRTK.Oculus/Packages/com.xrtk.oculus/Editor/OculusManifestPreprocessor.cs (hunk 3 above)
Comment (SimonDarksideJ) [reply XRTK.Oculus/Packages/com.xrtk.oculus/Editor/OculusManifestPreprocessor.cs]: Oculus source, can't change too much without invalidating the license

File: XRTK.Oculus/Packages/com.xrtk.oculus/Editor/OculusUpdaterStub.cs (hunk 2 above)
Comment (SimonDarksideJ) [reply XRTK.Oculus/Packages/com.xrtk.oculus/Editor/OculusUpdaterStub.cs]: Updated

**Comment by SimonDarksideJ** [comment]
Comments addressed where possible without affecting the license too much

**Pull request merged**